spec/requests/api/v0/articles_spec.rb

```diff
@@ -102,6 +102,49 @@ def json_response
     end
   end
 
+  describe "GET /api/articles/me" do
+    context "when request is unauthenticated" do
+      it "return unauthorized" do
+        get me_api_articles_path
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context "when request is authenticated" do
+      let_it_be(:user) { create(:user) }
+      let_it_be(:access_token) { create :doorkeeper_access_token, resource_owner: user }
+
+      it "works with bearer authorization" do
+        headers = { "authorization" => "Bearer #{access_token.token}", "content-type" => "application/json" }
+
+        get me_api_articles_path, headers: headers
+        expect(response.content_type).to eq("application/json")
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "return proper response specification" do
+        get me_api_articles_path, params: { access_token: access_token.token }
+        expect(response.content_type).to eq("application/json")
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "return only user's articles including markdown" do
+        create(:article, user: user)
+        create(:article)
+        get me_api_articles_path, params: { access_token: access_token.token }
+        expect(json_response.length).to eq(1)
+        expect(json_response[0]["markdown"]).not_to be_nil
+      end
+
+      it "supports pagination" do
+        stub_const("#{Api::V0::ArticlesController}::PER_PAGE", 2)
```

Comment:
In order to avoid creating 30 db objects, I had to make `per_page` a class instance, and I had to follow https://github.com/rspec/rspec-mocks/issues/1079#issuecomment-215620243 hack to properly stub it without overwriting class definition.

Reply:
If you add a `per_page` parameter you don't need the constant at all, just create 2 objects with `create_list` and ask for the second page

Reply:
updated